### PR TITLE
STORM-1937 Fix WindowTridentProcessor cause NullPointerException

### PR DESCRIPTION
**Pull request by wangperry**
I'm working with trident and try to use windowing  support, under the local model is fine, but in distributed mode we got the NullPointerException; This fixed the error in WindowTridentProcessor.java

### CODE REVIEW

File: storm-core/src/jvm/org/apache/storm/trident/windowing/WindowTridentProcessor.java [hunk 1]
```diff
@@ -163,11 +163,14 @@ public void finishBatch(ProcessorContext processorContext) {
         Iterable<Object> triggerValues = null;
 
         if (retriedAttempt(batchId)) {
-            pendingTriggerIds = (List<Integer>) windowStore.get(inprocessTriggerKey(batchTxnId));
-            for (Integer pendingTriggerId : pendingTriggerIds) {
-                triggerKeys.add(triggerKey(pendingTriggerId));
+            Object triggerIds = windowStore.get(inprocessTriggerKey(batchTxnId));
```
Comment (satishd) [inline storm-core/src/jvm/org/apache/storm/trident/windowing/WindowTridentProcessor.java]: @wangperry Can you remove creating one more variable. You need to wrap with a simple null check like below.

```java
if(pendingTriggerIds != null)  {
    for (Integer pendingTriggerId : pendingTriggerIds) {
      triggerKeys.add(triggerKey(pendingTriggerId));
   }
   triggerValues = windowStore.get(triggerKeys);
}
```

**Comment by wangperry** [comment]
Thanks @satishd

**Comment by wangperry** [comment]
Thanks @satishd

**Comment by satishd** [comment]
@wangperry Can you squash the commits?

File: storm-core/src/jvm/org/apache/storm/trident/windowing/WindowTridentProcessor.java [hunk 2]
```diff
@@ -164,10 +164,12 @@ public void finishBatch(ProcessorContext processorContext) {
 
         if (retriedAttempt(batchId)) {
             pendingTriggerIds = (List<Integer>) windowStore.get(inprocessTriggerKey(batchTxnId));
-            for (Integer pendingTriggerId : pendingTriggerIds) {
-                triggerKeys.add(triggerKey(pendingTriggerId));
+            if (pendingTriggerIds != null){
```
Comment (HeartSaVioR) [inline storm-core/src/jvm/org/apache/storm/trident/windowing/WindowTridentProcessor.java]: nit: space between ) and {

**Comment by HeartSaVioR** [comment]
@wangperry +1 except nit style comment. 
Please squash the commits into one, and I'd recommend you to create pull request against master branch, but if you mind to do I'll take care of it.

**Comment by wangperry** [comment]
@HeartSaVioR ,  I am newbie, could you help me squash multi commits into one? Thanks

**Comment by HeartSaVioR** [comment]
@wangperry Sorry I've occupied a bit. I can handle it so you don't need to worry. :) +1

**Comment by satishd** [comment]
+1

**Pull request closed without merging**